kubernetes/structures.go

```diff
@@ -118,19 +119,31 @@ func expandStringSlice(s []interface{}) []string {
 	return result
 }
 
-func flattenMetadata(meta metav1.ObjectMeta, d *schema.ResourceData, metaPrefix ...string) []interface{} {
+func flattenMetadata(meta metav1.ObjectMeta, d *schema.ResourceData, providerMetadata interface{}, metaPrefix ...string) []interface{} {
 	m := make(map[string]interface{})
 	prefix := ""
 	if len(metaPrefix) > 0 {
 		prefix = metaPrefix[0]
 	}
 	configAnnotations := d.Get(prefix + "metadata.0.annotations").(map[string]interface{})
-	m["annotations"] = removeInternalKeys(meta.Annotations, configAnnotations)
+
+	var ignoreAnnotations []string
+	if v, ok := providerMetadata.(kubeClientsets).configData.Get("ignore_annotations").([]interface{}); ok {
+		ignoreAnnotations = expandStringSlice(v)
+	}
+
+	m["annotations"] = removeInternalKeys(meta.Annotations, configAnnotations, ignoreAnnotations)
```

Comment:
Thanks, @jrhouston. Done!

I reverted function names back to their origins and added a new one `removeKeys` that works in the same way as `removeInternalKeys` but accepts one extra argument -- Kubernetes metadata(annotations and labels) keys that a practitioner wants to ignore. `removeKeys` also calls a new function `isIgnoteKey` instead of `isInternalKey` that verifies whether a Kubernetes object has metadata keys that should be ignored.

`removeKeys` seems to be more generic and actually can cover use cases of `removeInternalKeys`. Maybe we should think about just one function, but call it two times. The first time, to ignore "internal" keys that we can store in a constant as a list of regular expressions. The second time, to ignore keys that the practitioner wants to ignore.

Reply:
Thanks @arybolovlev. I agree, we could take the logic from `isInternalKey` and turn it into a const of regular expressions that we can then merge with the user supplied list to pass into a single `removeKeys()` call. Let's leave that for another PR though.